api/envoy/extensions/filters/http/jwt_authn/v3/config.proto

```diff
@@ -230,6 +230,18 @@ message JwtProvider {
   //
   string payload_in_metadata = 9;
 
+  // Alters the payload representation in the request dynamic metadata to facilitate its use in matching.
+  message NormalizePayload {
+    // Converts the space delimited claims into lists of string, e.g. ``"email
+    // profile"`` is translated to ``["email", "profile"]``. For example,
+    // ``scope`` claim requires this normalization per `RFC8693
+    // <https://datatracker.ietf.org/doc/html/rfc8693#name-scope-scopes-claim>`_.
+    repeated string space_delimited_claims = 1;
```

Comment:
I think I follow the idea of the normalization, but I'm' a bit confused about how this works. It seems this is "configuration for how the JWT filter normalizes", but then this is a repeated string of some kind with the actual claims?

Reply:
Yeah, the idea is to normalize claims into dynamic metadata, so a jwt with claim `"email profile"` becomes dynamic metadata like:

```yaml
filter_metadata:
  envoy.jwt_authn:
    claims:
    - email
    - profile
```

Later on, a route matcher specifies:
```yaml
match:
  dynamic_metadata:
  - filter: envoy.jwt_authn
    path:
    - key: claims
    value:
      list_match:
        one_of:
          string_match: {"exact": "email"}
```

to select routes with JWT token that contains "email".

This is how it works in Istio with a help from a custom filter `istio_authn` that performs this normalization into a separate dynamic metadata.

Reply:
Sorry, still a bit dense. Is `"email profile"` in the `JwtProvider` or in the JWT claim or both?

Reply:
It's in the JWT claim. For example, in https://datatracker.ietf.org/doc/html/rfc8693#name-scope-scopes-claim, they give this example:
```
 {
   "aud":"https://consumer.example.com",
   "iss":"https://issuer.example.com",
   "exp":1443904177,
   "nbf":1443904077,
   "sub":"dgaf4mvfs75Fci_FL3heQA",
   "scope":"email profile phone address"
 }
```

I don't like this encoding with white spaces, but it appears to be common due to auth0 adopting it (https://auth0.com/docs/get-started/apis/scopes/sample-use-cases-scopes-and-claims) among others.

Reply:
An alternative might be to make this option into a boolean `convert_space_delimited_scope`.

Reply:
Oh, so `space_delimited_claims` is a list of claims in which each claim should be converted from space delimited to list, right?

In the case I'd change the wording slightly:

```
Each claim in this list will be interpreted as a space-delimited string and converted to a list of strings based on the delimited....
```

Reply:
Thanks, updated the wording.